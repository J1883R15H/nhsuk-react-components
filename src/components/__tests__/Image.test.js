import React from 'react';
import { shallow } from 'enzyme';
import Image from '../image';

describe('Image', () => {
  it('matches snapshot and element', () => {
    const shallowMount = shallow(<Image src="imagesrc" alt="imagealt" />);
    expect(shallowMount).toMatchSnapshot();
    expect(
      shallowMount.containsMatchingElement(
        <figure className="nhsuk-image">
          <img className="nhsuk-image__img" alt="imagealt" src="imagesrc" />
        </figure>
      )
    ).toBeTruthy();
    shallowMount.unmount();
  });
  it('renders caption', () => {
    const shallowMount = shallow(
      <Image src="imagesrc" alt="imagealt" caption="Caption" />
    );
    expect(
      shallowMount.containsMatchingElement(
        <figcaption className="nhsuk-image__caption">Caption</figcaption>
      )
    ).toBeTruthy();
    shallowMount.unmount();
  });
});
