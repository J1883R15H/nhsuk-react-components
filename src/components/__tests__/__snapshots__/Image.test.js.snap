// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image matches snapshot and element 1`] = `
<figure
  className="nhsuk-image"
  style={Object {}}
>
  <img
    alt="imagealt"
    className="nhsuk-image__img"
    src="imagesrc"
  />
</figure>
`;
